.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,21 +1,32 @@
-> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.
+# Add here a title for your PR
 
-### What does this PR do?
+<!-- 
+Please use one of the following prefixes for the title:
+docs: Documentation not including procedures. Engineering review is mandatory.
+procedures: Documentation including procedures. Testing procedures is mandatory. Engineering and QE review is mandatory (Engineering can review on behalf of QE). 
+chore: Choreography, release, tooling, version upgrades.
+fix: Fix build, language, links, or metadata.
+-->
 
+<!-- Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR. -->
 
-### What issues does this PR fix or reference?
+## What does this PR do?
 
+## What issues does this PR fix or reference?
 
-### Specify the version of the product this PR applies to.
+## Specify the version of the product this PR applies to
 
+## Pull Request Checklist
 
-### PR Checklist
+The author and the reviewers validate the content of this Pull Request with this checklist, in addition to the [automated tests](code_review_checklist.adoc).
 
-As the author of this Pull Request I made sure that:
-
-- [ ] `vale` has been run successfully against the PR branch
-- [ ] Link checker has been run successfully against the PR branch
-- [ ] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
-- [ ] Changed article references are updated where they are used (or a redirect has been configured on the docs side):
+- Any procedure:
+  - [ ] Successfully tested.
+  - [ ] Describes a scenario already covered by QE tests, otherwise Che QE team has acknowledged an issue.
+- Any page or link rename:
+  - [ ] The page contains a redirection for the previous URL.
+  - Propagate the URL change in:
     - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/src/services/bootstrap/branding.constant.ts)
     - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
+- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
+- [ ] *`Validate language on files added or modified`* step reports no vale warnings.
```

Comment:
To my mind this requirement is not fair. If one changes a line in a big file, this will force that person to review whole page and then, after fixing warnings / mistakes, retest the flow and create issues for QE... overkill for changing a single line...

Reply:
I had another experience with `vale`: it gave warnings for just using some words or phrases. But sometimes they are just necessary. I had to contact a person from docs team and usually it ended up with justification of the phrase using in that specific place...

Reply:
We have already fixed all vale `errors`. We have currently only 80 vale `warnings`, I believe that targeting to fix the remaining warnings is a reachable goal. We have more than 4000 `suggestions`, and new suggestions don't block PR.

Staying compliant with a style guide is hard, yes. It's like with any linter for code: it defines a very normative way of writing, and it doesn't let room for editorial disagreements about interpretation of the style guide.

Reply:
> Staying compliant with a style guide is hard, yes. It's like with any linter for code: it defines a very normative way of writing, and it doesn't let room for editorial disagreements about interpretation of the style guide.

I think they are different, totally. When a linter forces developers to comply with code style it does not prevent from using language operators inside code. But vale does when says that specific word or phrase is not allowed.
I don't know how to properly resolve this on technical level: to not to block and to not to allow unnecesary usage of the phrases. The best way, seems, is to have human review instead of blocking by vale.

Reply:
Scope of a linter

Some code linters go much deeper than indentation. Vale is very comparable to ansible-lint. It has rules to let discourage the user to use anti-patterns. From my experience, it's an incredible tool to increase the quality and consistency of Ansible playbooks, and to help new contributors learn the good practices. "As any other linter, it is opinionated." says the doc. See the rules: https://ansible-lint.readthedocs.io/en/latest/rules.html

Blocking vs. non-blocking

Current situation:

* 0 errors. Blocks the merge, since months. Time to document it.
* 80 warnings. Don't block the merge. Now that we have reached the zero errors target, the brief for reviewers is to get rid of warnings. I believe it's good to share this information in the open.
* Thousands of suggestions. Don't block merge. Need human review. Nothing changing.

As we have lived with Vale in not blocking mode for some months, I can enumerate some disadvantages:

* "check passed" didn't mean the content of the PR is fine, reviewer needed to deep dive in the logs
* more wait time for the contributor, as the review needs more work
* increasing number of errors and warnings merged
* more post-merge fixes

Now we have lived with Vale in blocking mode for some months,  can also enumerate some advantages:

Advantages of having vale block a merge for the contributor:
* reduced wait time on entry level review
* less frictions

Advantages of having vale block a merge for the reviewers:
* confidence in the "check passed" status
* less mistakes needing a post-merge fix
* less frictions
* less repetitive tasks
* more time for complex tasks: test the procedures, ensure compliance with more complex languages rules.

Rules accuracy

There is a given: we must comply to the IBM Style Guide and the Supplementary Style Guide. That's not something open to discussion.

Vale implements a subset of these rules, which are really the basics, and we can discuss the implementation in this repository:

* Severity of a rule: error should be warning or suggestion? 
* Refine a rule.

I am all for that. The rules we are using are need maturation, but they are already in a very good shape. 

But let's discuss on each rule in particular.

Reply:
```suggestion
- [ ] As much as possible, *`Validate language on files added or modified`* step reports no vale warnings.
```